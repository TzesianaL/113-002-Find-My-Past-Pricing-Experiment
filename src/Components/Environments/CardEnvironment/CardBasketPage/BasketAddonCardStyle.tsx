import { PriceStyle } from 'Components/Shared/Price/PriceStyle'
import { TitleStyle } from 'Components/Shared/Title/TitleStyle'
import styled from 'styled-components'

/**
 * Styled component is used to decorate CardWrapper component.
 * Props are passed from ThemeProvider component.
 */

export const BasketAddonCardStyle = styled.div`
  box-shadow: 0 2px 4px 0 rgb(0 0 0 / 12%);
  border-radius: 5px;
  margin: 0em 1em 1em 1em;
  background-color: white;
  align-items: center;
  display: flex;
  justify-content: center;
  flex-direction: column;
  width: 97%;
  color: black;
`

export const BasketAddonCardRowStyle = styled.div`
  background-color: white;
  display: flex;
  width: 89%;
  // margin: 0.5em 1em 1em 1em;
  justify-content: space-between;
  flex-direction: row;
  color: black;

  ${TitleStyle} {
    color: black;
  }
  ${PriceStyle} {
    color: black;
    margin: 0.2em 0em 0.2em 0em;
  }
`
