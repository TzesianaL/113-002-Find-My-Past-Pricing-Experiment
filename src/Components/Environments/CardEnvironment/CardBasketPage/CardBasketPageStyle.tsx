import styled from 'styled-components'
import { BasketCardWrapper } from './BasketCard/BasketCardStyle'

export const CardBasketPageStyle = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-bottom: 3em;

  ${BasketCardWrapper} {
    margin-top: 2em;
  }

  .introText {
    font-size: 1.2em;
    max-width: 50em;
    @media screen and (max-width: 600px) {
      max-width: 17em;
    }
  }

  .footerText {
    font-size: 1.2em;
    max-width: 50em;
  }
`

export const ButtonContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  button {
    margin: 0.5em 1em;
  }
`
