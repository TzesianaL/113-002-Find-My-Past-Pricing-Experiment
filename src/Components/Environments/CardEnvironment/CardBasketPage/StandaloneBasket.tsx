import React, { FC } from 'react'
import { useSelector } from 'react-redux'
import {
  getAddonsInBasket,
  getFirstBasketItem,
} from 'redux/reducers/logicStore/logicStore.selectors'
import { BasketCardWrapper } from './BasketCard/BasketCardStyle'
import { BasketPageStyle } from '../../../Shared/BasketPage/BasketPageStyle'
import { BasketSummary } from './BasketSummary/BasketSummary'

export const StandaloneBasket: FC = () => {
  const basketItem = useSelector(getFirstBasketItem)
  const addons = useSelector(getAddonsInBasket)

  return (
    <BasketPageStyle>
      <BasketCardWrapper>
        <BasketSummary
          item={basketItem}
          addons={addons}
          prePriceText={basketItem.basketOptions?.prePriceText}
          postPriceText={basketItem.basketOptions?.postPriceText}
        />
      </BasketCardWrapper>
    </BasketPageStyle>
  )
}
