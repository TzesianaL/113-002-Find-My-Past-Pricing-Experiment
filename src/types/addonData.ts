import { Periods } from './periods'

export interface AddonData {
  id: string
  title: string
  subtitle?: string
  features?: string[]
  price?: number
  pricePeriod?: Periods
  includedProductIDs?: string[]
  excludedProductIDs?: string[]
  mutex?: boolean
  group?: number
  mutexGroup4?: boolean
  mutexGroup2?: boolean
}
