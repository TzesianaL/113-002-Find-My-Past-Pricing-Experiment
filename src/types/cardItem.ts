import { Periods } from './periods'

interface BasketCardOptions {
  prePriceText?: string
  postPriceText?: string
}

export type ContractDuration = '24 months' | '30 days'

// Type guard function for runtime validation of ContractDuration
export const isContractDuration = (
  value: string
): value is ContractDuration => {
  return value === '24 months' || value === '30 days'
}

export interface CardItem {
  id: string
  title: string
  subtitle: string
  subtitle2?: string
  price: number
  currency?: string
  priceString?: string
  features: string[]
  cardColor?: string
  highlightedText?: string
  headerImage?: string
  upfrontPrice?: number
  basketOptions?: BasketCardOptions
  pricePeriod?: Periods
  contractDuration?: ContractDuration
  parentSpeedId?: string
}

export interface SpeedCardItem {
  title: string
  subtitle: string
  price: number
  currency?: string
  highlightedText?: string
  id: string
  pricePeriod?: Periods
  purchaseButtonText?: string
  mainSpeed: string
  downloadSpeed: string
  downloadExample: string
  downloadTime: string
  uploadSpeed: string
  uploadExample: string
  uploadTime: string
  speed?: number
}
