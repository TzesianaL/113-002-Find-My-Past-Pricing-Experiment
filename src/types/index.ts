export * from './addonData'
export * from './cardItem'
export * from './layout'
export * from './periods'
export * from './reduxData'
export * from './splashItem'
export * from './pcw'
export * from './environmentType'
