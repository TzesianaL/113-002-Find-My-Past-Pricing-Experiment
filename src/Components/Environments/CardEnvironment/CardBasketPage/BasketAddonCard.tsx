import React, { FC } from 'react'
import Price from 'Components/Shared/Price/Price'
import { AddonData, Periods } from 'types'
import Title from 'Components/Shared/Title/Title'
import { convertPricePeriods } from 'functions/functions'
import {
  BasketAddonCardStyle,
  BasketAddonCardRowStyle,
} from './BasketAddonCardStyle'

/**
 * AddonCardProps contains all appropriate props we should pass to nested components.
 */
export interface BasketAddonCardProps {
  selectedAddons: AddonData[]
  convertedPricePeriod?: Periods
}

/**
 * BasketAddonCard is used to wrap all components we render in order to display all selected addons.
 * ThemeProvider component is used. By using the theme object we provide same styling options to all components.
 */
const BasketAddonCard: FC<BasketAddonCardProps> = ({
  selectedAddons,
  convertedPricePeriod,
}) => {
  return (
    <BasketAddonCardStyle>
      {/* <Title>Add Ons</Title> */}
      {selectedAddons.map((selectedAddon) => (
        <BasketAddonCardRowStyle key={selectedAddon.id}>
          <Title subtitle style={{ width: '300px' }}>
            {selectedAddon.title}
          </Title>
          {selectedAddon.price !== undefined && (
            <Price
              style={{ width: '120px' }}
              price={convertPricePeriods(
                selectedAddon.price,
                selectedAddon.pricePeriod,
                convertedPricePeriod
              )}
              pricePeriod="month"
              id={selectedAddon.title}
            />
          )}
        </BasketAddonCardRowStyle>
      ))}
    </BasketAddonCardStyle>
  )
}

export default BasketAddonCard
