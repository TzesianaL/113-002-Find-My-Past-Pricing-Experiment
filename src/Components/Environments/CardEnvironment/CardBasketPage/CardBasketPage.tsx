import React, { FC, useEffect } from 'react'
import { useDispatch } from 'react-redux'
import { merge } from 'lodash-es'
import { ThemeProvider } from 'styled-components'
import { CardItem, AddonData, Periods } from 'types'
import { addOutput, OutputTypes } from 'redux/reducers/output/outputSlice'
import BasketPage from 'Components/Shared/BasketPage/BasketPage'
import { BasketCardWrapper } from './BasketCard/BasketCardStyle'
import { BasketSummary } from './BasketSummary/BasketSummary'

/**
 *
 */
export interface BasketPageProps {
  bannerName?: string
  bannerLogoSrc?: string
  introText?: string
  footerText?: string
  prePriceText?: string
  postPriceText?: string
  checkOutButtonText?: string
  emptyBasketButtonText?: string
  item: CardItem
  showCancelButton?: boolean
  onCheckout: () => void
  onEmptyBasket: () => void
  onCancel?: () => void
  cancelButtonText?: string
  addons: AddonData[]
  convertedPricePeriod?: Periods
}

const CardBasketPage: FC<BasketPageProps> = ({
  bannerLogoSrc,
  bannerName,
  prePriceText,
  postPriceText,
  emptyBasketButtonText,
  introText,
  footerText,
  item,
  showCancelButton,
  onCancel,
  onCheckout,
  onEmptyBasket,
  checkOutButtonText,
  cancelButtonText,
  addons,
  convertedPricePeriod,
}) => {
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(
      addOutput({
        key: OutputTypes.selectedAddons,
        value: addons?.map((addon) => addon.id).toString(),
      })
    )
    dispatch(
      addOutput({
        key: OutputTypes.addonPrice,
        value: addons?.map((addon) => addon.price).toString(),
      })
    )
  }, [dispatch, addons])

  return (
    <ThemeProvider
      key={item.id}
      theme={(theme) => {
        return merge({}, theme, {
          card: {
            backgroundColor: item.cardColor,
          },
        })
      }}
    >
      <BasketPage
        bannerLogoSrc={bannerLogoSrc}
        bannerName={bannerName}
        introText={introText}
        footerText={footerText}
        showCancelButton={showCancelButton}
        onCancel={onCancel}
        onCheckout={onCheckout}
        onEmptyBasket={onEmptyBasket}
        emptyBasketButtonText={emptyBasketButtonText}
        checkOutButtonText={checkOutButtonText}
        cancelButtonText={cancelButtonText}
        basketContents={
          <BasketCardWrapper>
            <BasketSummary
              item={item}
              addons={addons}
              convertedPricePeriod={convertedPricePeriod}
              prePriceText={prePriceText}
              postPriceText={postPriceText}
            />
          </BasketCardWrapper>
        }
      />
    </ThemeProvider>
  )
}

export default CardBasketPage
