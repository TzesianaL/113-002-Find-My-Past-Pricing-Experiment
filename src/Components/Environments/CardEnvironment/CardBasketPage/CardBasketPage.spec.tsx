import React from 'react'
import 'jest-styled-components'
import { CardItem, AddonData, DataStateType, BlockEnums } from 'types'
import configureStore from 'redux-mock-store'
import { screen, render, fireEvent } from '@testing-library/react'
import { Provider } from 'react-redux'
import { testItems } from 'stories/CardEnvironment/__fixtures__/card.fixtures'
import CardBasketPage from './CardBasketPage'

const mockStore = configureStore([])

describe('<BasketPage/>', () => {
  const cardToAdd: CardItem = {
    title: 'test card',
    subtitle: '',
    id: 'TESTCARD',
    price: 5.0,
    features: [],
    cardColor: '',
  }

  const basicDataStore: DataStateType = {
    dataLoaded: true,
    storeAddons: [],
    pcwData: [],
    pcwSchema: [],
    addonLayout: { blockType: BlockEnums.Null },
    condition: 0,
    splash: {
      title: '',
      items: [],
    },
    storeItems: [],
    storeLogoSrc: '',
    storeName: '',
    theme: {},
    storePageOptions: {},
    basketPageOptions: {},
  }
  let store

  const testAddon: AddonData[] = [
    {
      id: '1',
      title: 'Trauma Cover',
      subtitle: 'What does it include?',
      price: 19.99,
    },
  ]

  beforeEach(() => {
    store = mockStore({
      logic: {
        basket: [cardToAdd.id],
        itemOrder: [],
      },
      data: { ...basicDataStore, storeItems: [cardToAdd] },
      output: {},
    })
  })

  // A test harness for the BasketPage prop
  // Using this makes the tests clearer as it removes the boiler plate
  // that BasketPage needs, but we don't really care about
  // We can also override any of the props on the basket page as well
  const TestBasketPage = (props) => (
    <Provider store={store}>
      <CardBasketPage
        bannerLogoSrc=""
        bannerName="Store Name"
        item={cardToAdd}
        onCheckout={() => {}}
        onEmptyBasket={() => {}}
        {...props}
      />
    </Provider>
  )

  it('should render correctly', () => {
    const { asFragment } = render(<TestBasketPage />)
    expect(asFragment()).toMatchSnapshot()

    expect(screen.getByText(cardToAdd.title)).toBeInTheDocument()

    const expectedTotalPrice = new Intl.NumberFormat('en-GB', {
      style: 'currency',
      currency: 'GBP',
    }).format(cardToAdd.price || 0)

    expect(screen.getByText(expectedTotalPrice)).toBeInTheDocument()
  })

  it('should show the provided checkOutButtonText on the Check Out Button', () => {
    const testLabel = 'CheckOut Test'

    const button = render(
      <TestBasketPage checkOutButtonText={testLabel} />
    ).getByRole('button', { name: testLabel })
    expect(button).toBeInTheDocument()
  })

  it('should show the provided emptyBasketButtonText on the Empty Basket Button', () => {
    const testLabel = 'Empty Basket Test'
    const button = render(
      <TestBasketPage emptyBasketButtonText={testLabel} />
    ).getByRole('button', { name: testLabel })

    expect(button).toBeInTheDocument()
  })

  it("should render the default value of 'Check Out' when checkOutButtonText isn't provided", () => {
    const button = render(
      <Provider store={store}>
        <CardBasketPage
          bannerLogoSrc="test"
          bannerName="test"
          item={testItems[0]}
          onCheckout={() => {}}
          onEmptyBasket={() => {}}
          addons={testAddon}
        />
      </Provider>
    ).getByRole('button', { name: /Check Out/ })
    expect(button).toBeInTheDocument()
  })

  it("should render the default value of 'Empty Basket' when emptyBasketButtonText isn't provided", () => {
    const button = render(<TestBasketPage />).getByRole('button', {
      name: /Empty Basket/,
    })
    expect(button).toBeInTheDocument()
  })

  it("should render 'Cancel Button' when showCancelButton is true with the display default button text 'Cancel', and using the onCancel function ", () => {
    const cancelFunction = jest.fn()
    const button = render(
      <TestBasketPage showCancelButton onCancel={cancelFunction} />
    )
    expect(button.getByRole('button', { name: /Cancel/ })).toBeInTheDocument()

    fireEvent.click(screen.getByText(/Cancel/i))
    expect(cancelFunction).toHaveBeenCalledTimes(1)
  })

  it("should not render 'Cancel' Button when showCancelButton property is not provided ", () => {
    const button = render(<TestBasketPage />)
    expect(button.queryByText(/Cancel/)).toBeFalsy()
  })

  it('should throw an Error when showCancelButton is set to true and onCancel function is not provided', () => {
    const mockError = jest.spyOn(console, 'error')
    mockError.mockImplementation(() => {})

    expect(() => {
      render(<TestBasketPage showCancelButton />)
    }).toThrowError(
      'OnCancel function is undefined when showCancelButton is provided'
    )

    mockError.mockRestore()
  })

  it('should call onCancel function,when showCancelButton is true and make sure it is clicked ', () => {
    const mockedFunction = jest.fn()
    render(<TestBasketPage showCancelButton onCancel={mockedFunction} />)
    fireEvent.click(screen.getByText(/Cancel/i))
    expect(mockedFunction).toHaveBeenCalledTimes(1)
  })

  it('should render pricePeriod when it is provided', () => {
    store = mockStore({
      logic: {
        basket: [cardToAdd.id],
        itemOrder: [],
      },
      data: {
        ...basicDataStore,
        storeItems: [{ ...cardToAdd, pricePeriod: 'year' }],
      },
      output: {},
    })

    // pricePeriod is provided on testitems[1]
    render(
      <Provider store={store}>
        <CardBasketPage
          bannerLogoSrc="test"
          bannerName="test"
          item={{
            ...testItems[1],
          }}
          addons={testAddon}
          onCheckout={() => {}}
          onEmptyBasket={() => {}}
        />
      </Provider>
    )

    expect(screen.getByText(/per year/)).toBeInTheDocument()
  })

  it('should not render pricePeriod when it is not provided', () => {
    const el = render(<TestBasketPage />)
    expect(
      el.queryByText(
        /per day/ ||
          /per week/ ||
          /per fortnight/ ||
          /per month/ ||
          /per quarter/ ||
          /per year/
      )
    ).toBeFalsy()
  })

  it('should show the provided cancelButtonText on the Cancel Button', () => {
    const testLabel = 'Cancel Button Text-Test'
    const mockedFunction = jest.fn()
    const el = render(
      <TestBasketPage
        cancelButtonText={testLabel}
        showCancelButton
        onCancel={mockedFunction}
      />
    ).getByRole('button', {
      name: /Cancel Button Text-Test/,
    })
    expect(el).toBeInTheDocument()
  })

  it("should render the default value of Cancel when cancelButtonText isn't provided", () => {
    const mockedFunction = jest.fn()
    const button = render(
      <TestBasketPage showCancelButton onCancel={mockedFunction} />
    ).getByRole('button', { name: /Cancel/ })
    expect(button).toBeInTheDocument()
  })
})
