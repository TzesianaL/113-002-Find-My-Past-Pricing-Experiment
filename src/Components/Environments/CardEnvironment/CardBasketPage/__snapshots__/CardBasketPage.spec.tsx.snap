// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasketPage/> should render correctly 1`] = `
<DocumentFragment>
  .c11 {
  color: #ffffff;
  background-color: #003D59;
  cursor: pointer;
  font-size: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0px;
  border-radius: 6px;
  padding: 0.7em 1em;
}

.c11:active {
  background-color: #336D89;
}

.c1 {
  background-color: #6300AA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fill-available;
  width: -moz-available;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white !important;
  padding: 0.6rem 0 0.6rem 0;
}

.c1 p {
  width: 100%;
  font-size: rem;
}

.c1 img {
  padding: 0 1em;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 3em;
}

.c0 .introText {
  font-size: 1em;
  max-width: 50em;
}

.c0 .footerText {
  font-size: 1em;
  max-width: 50em;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: auto;
  width: 40%;
  margin-top: 3em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .cardWrapper {
  width: 75%;
}

.c10 {
  margin-top: 1em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 button {
  margin: 0.5em 1em;
}

.c7 {
  text-align: center;
  text-transform: uppercase;
  font-weight: bold;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.2em;
  margin: 0.5em 0em 0.5em 0em;
}

.c8 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #ffffff;
  font-size: 1.2em;
  margin: 0.5em 0em 0.5em 0em;
}

.c5 {
  background-color: #D8D2E8;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  padding: 16px 1em 24px 1em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 2px;
  border-color: #000000;
  border-radius: 1.5em;
  max-width: 100%;
  color: #000000;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.12);
  margin: 0em 1em 1em 1em;
  font-size: 0.8em;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #000000;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1em 0;
  margin: 1em 0;
  border: 1px solid black;
  max-width: 50em;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c9 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  font-weight: bold;
  color: #000000;
}

.c9 div {
  margin: 0;
  padding: 0em 1em 0em 1em;
  text-align: right;
}

.c9 h3 {
  margin: 0;
  padding: 1em;
  text-align: right;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 img {
  height: 36px;
  padding: 0 0.25em;
}

@media (max-width:600px) {
  .c2 {
    width: 100%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    margin: auto;
    margin-top: 3em;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <p>
        Store Name
      </p>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h1>
          Your Basket
        </h1>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <div
              class="c6"
            >
              <div
                class="c7"
              >
                <div>
                  test card
                </div>
              </div>
            </div>
            <div
              class="c8"
            >
              <div />
            </div>
          </div>
        </div>
        <div
          class="c9"
        >
          <h3>
            Total Price:
            <span>
              Â£5.00
            </span>
          </h3>
        </div>
      </div>
      <div
        class="c10"
      >
        <button
          class="c11"
        >
          Empty Basket
        </button>
        <button
          class="c11"
        >
          Check Out
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
